cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/SpannerBrowser/DatabaseList/index.tsx

```diff
@@ -0,0 +1,174 @@
+
+    listSpannerDatabases(connectionId, instanceId);
+  }
+
+  public render() {
+    if (this.props.loading) {
+      return <LoadingSVGCentered />;
+    }
+
+    const {databaseList} = this.props;
+    const connectionId = this.props.connectionId || objectQuery(this.props, 'match', 'params', 'connectionId');
+    const instanceId = this.props.instanceId || objectQuery(this.props, 'match', 'params', 'instanceId');
+
+    if (!databaseList.length) {
+      return (
+        <div className="empty-search-container">
+          <div className="empty-search text-xs-center">
+            <strong>
+              {T.translate(`${PREFIX}.EmptyMessage.emptyDatabaseList`, {
+                connectionName: connectionId,
+                instanceName: instanceId,
+              })}
+            </strong>
+          </div>
+        </div>
+      );
+    }
+
+    const Tag = this.props.enableRouting ? Link : 'span';
+    const namespace = getCurrentNamespace();
+    const path = `/ns/${namespace}/connections/spanner/${connectionId}`;
+
+    return (
+      <div className="list-view-container">
+        <div className="sub-panel">
+          <div className="path">
+            <Tag
+              to={path}
+              className="instance-path"
+              onClick={listSpannerInstances.bind(null, connectionId)}
+            >
+              {T.translate(`${PREFIX}.instances`)}
+            </Tag>
+            <span> / </span>
+            <span>{instanceId}</span>
+          </div>
+          <div>
+            {T.translate(`${PREFIX}.databaseCount`, {context: databaseList.length})}
+          </div>
+        </div>
+
+        <div className="list-table">
+          <div className="table-header">
+            <div className="row">
+              <div className="col-xs-12">
+                {T.translate(`${PREFIX}.name`)}
+              </div>
+            </div>
+          </div>
+
+          <div className="table-body">
+            {
+              databaseList.map((database: ISpannerDatabaseObject) => {
+                const ElemTag = this.props.enableRouting ? Link : 'div';
+                const instanceUrl = `/ns/${namespace}/connections/spanner/${connectionId}/instances/${instanceId}`;
+                const databaseUrl = `${instanceUrl}/databases/${database.name}`;
+
+                return (
+                  <ElemTag
+                    key={database.name}
+                    to={databaseUrl}
+                    onClick={this.clickHandler.bind(null, database.name)}
+                  >
+                    <div className="row content-row">
+                      <div className="col-xs-12">
+                        <IconSVG name="icon-database" />
+                        {database.name}
+                      </div>
+                    </div>
+                  </ElemTag>
+                );
+              })
+            }
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
```

Comment:
nit: Can we set the type for return object? we know it will be `ISpannerDatabaseListViewProps` right?